src/index.ts

```diff
@@ -248,6 +256,42 @@ export const supportedV2RouteHandler = (req, res, next, server) => {
   return server(req, res, next)
 }
 
+// Middleware that must be mounted on our GraphQL route _before_
+// our regular GraphQL handler. This is so we can register a new `res.end`
+// method that will intercept the response and check if any email provider
+// rate limting occurred.
+//
+// If so, we return a 429 status code and a JSON error message.
+app.use("/v2", function (req, res, next) {
+  const userAgent = req.headers["user-agent"] || ""
+  if (!userAgent.includes("Braze")) {
+    return next()
+  }
+
+  const originalEnd = res.end
+
+  res.end = function (...args) {
+    res.end = originalEnd
+    const [chunk, encoding, cb] = args
+
+    const contentType = res.get("content-type") || "text/html"
+
+    if (contentType.indexOf("application/json") === 0) {
+      const response = JSON.parse(chunk)
+
+      if (response.extensions && response.extensions.emailProviderLimited) {
+        res.status(429).json({ error: "rate limit reached, try again later" })
```

Comment:
This drops the other response data (including the extensions data), and just returns this tiny error blob, right?

Reply:
yup